src/components/Swap.tsx

```diff
@@ -57,9 +58,29 @@ const Swap = () => {
   const toPurse = useAtomValue(toPurseAtom);
   const offers = useAtomValue(offersAtom);
   const [swapButtonStatus, setSwapButtonStatus] = useAtom(swapButtonStatusAtom);
+  const instanceIds = useAtomValue(instanceIdsAtom);
 
-  const assetsLoaded =
-    brandToInfo.size && metrics && governedParams && instanceId;
+  const anchorPetnames = [...instanceIds.keys()];
+  const areAnchorsLoaded =
+    anchorPetnames.length &&
+    anchorPetnames
+      .map(petname => {
```

Comment:
no need to check length
```suggestion
  const anchorPetnames = [...instanceIds.keys()];
  const areAnchorsLoaded = anchorPetnames.map(petname => {
```